tests/php/sync/test_class.jetpack-sync-queue.php

```diff
@@ -52,13 +52,25 @@ function test_items_exist() {
 	}
 
 	function test_queue_lag() {
-		// lag is the difference in time between the age of the oldest item and the current time
-		$this->queue->reset();
-		$this->queue->add( 'foo' );
-		sleep( 3 );
-		$this->queue->add( 'bar' );
-		sleep( 3 );
-		$this->assertEquals( 6, intval( $this->queue->lag() ) );
+		/* @var $queue Jetpack_Sync_Queue|\PHPUnit\Framework\MockObject\MockObject */
+		$queue = $this->getMockBuilder( 'Jetpack_Sync_Queue' )
+			->setMethods( array( 'generate_option_name_timestamp' ) )
+			->setConstructorArgs( array( 'my_queue' ) )
+			->getMock();
+
+		$queue->expects( $this->at( 0 ) )
+			->method( 'generate_option_name_timestamp' )
+			->will( $this->returnValue( '1.5' ) );
+
+		$queue->expects( $this->at( 1 ) )
+			->method( 'generate_option_name_timestamp' )
+			->will( $this->returnValue( '3.0' ) );
+
+		$queue->reset();
+		$queue->add( 'foo' );
+		$queue->add( 'bar' );
+
+		$this->assertEquals( 6, intval( $queue->lag( 7.5 ) ) );
```

Comment:
Can we add a comment here 
Something along the lines of 
7.5 - 1.5 = 6.0 
or 
Current time - the first item timestamp = current lag. 
so that things are a bit more clear. :) I had to think about this part a bit.